classes/submission/Repository.php

```diff
@@ -520,16 +520,17 @@ public function canEditPublication(int $submissionId, int $userId): bool
 
         $submission = $this->get($submissionId);
         // any published or scheduled then probe
-        $hasLockedPublication = $submission
-            && $submission->getData('publications')
-                ->contains(fn($p) =>
+        $hasLockedPublication = $submission?->getData('publications')
+            ?->contains(
```

Comment:
If **only** `$submission` needs to be null-checked, then you don't need to add the `?->` to all subsequent members (e.g. `$submission?->a?->b?->c?->d`)

```suggestion
            ->contains(
```